aspnetcore/security/authorization/secure-data.md

```diff
@@ -14,305 +15,328 @@ uid: security/authorization/secure-data
+Services using Entity Framework Core must be registered for [dependency injection](xref:fundamentals/dependency-injection) using [AddScoped](/aspnet/core/api/microsoft.extensions.dependencyinjection.servicecollectionserviceextensions). The `ContactIsOwnerAuthorizationHandler` uses ASP.NET Core [Identity](xref:security/authentication/identity), which is built on Entity Framework Core. Register the handlers with the service collection so they're available to the `ContactsController` through [dependency injection](xref:fundamentals/dependency-injection). Add the following code to the end of `ConfigureServices`:
 
-[!code-csharp[Main](secure-data/samples/final/Startup.cs?name=AuthorizationHandlers)]
+[!code-csharp[Main](secure-data/samples/final2/Startup.cs?name=ConfigureServices&highlight=41-)]
 
 `ContactAdministratorsAuthorizationHandler` and `ContactManagerAuthorizationHandler` are added as singletons. They're singletons because they don't use EF and all the information needed is in the `Context` parameter of the `HandleRequirementAsync` method.
 
-The complete `ConfigureServices`:
+## Support authorization
 
-[!code-csharp[Main](secure-data/samples/final/Startup.cs?name=ConfigureServices)]
+In this section, you update the Razor Pages and add an operations requirements class.
 
-## Update the code to support authorization
+### Review the contact operations requirements class
 
-In this section, you update the controller and views and add an operations requirements class.
+Review the `ContactOperations` class. This class contains the requirements the app supports:
 
-### Update the Contacts controller
+[!code-csharp[Main](secure-data/samples/final2/Authorization/ContactOperations.cs)]
 
-Update the `ContactsController` constructor:
+### Create a base class for the Razor Pages
 
-* Add the `IAuthorizationService` service to  access to the authorization handlers. 
-* Add the `Identity` `UserManager` service:
+Create a base class that contains the services used in the contacts Razor Pages. The base class puts that initialization code in one location:
 
-[!code-csharp[Main](secure-data/samples/final/Controllers/ContactsController.cs?name=snippet_ContactsControllerCtor)]
+[!code-csharp[Main](secure-data/samples/final2/Pages/Contacts/DI_BasePageModel.cs)]
 
-### Add a contact operations requirements class
+The preceding code:
 
-Add the `ContactOperations` class to the *Authorization* folder. This class  contain the requirements our app supports:
-
-[!code-csharp[Main](secure-data/samples/final/Authorization/ContactOperations.cs)]
+* Adds the `IAuthorizationService` service to access to the authorization handlers.
+* Adds the Identity `UserManager` service.
+* Add the `ApplicationDbContext`.
 
 ### Update Create
```

Comment:
I suggest changing this title to align with the one below it. How about "Update the CreateModel"?